apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/Query.cs

```diff
@@ -33,13 +43,415 @@ public class Query
 
         internal string ParentPath { get; }
 
-        private readonly StructuredQuery _query;
+        // Visible for testing
+        internal StructuredQuery QueryProto { get; }
 
         internal Query(FirestoreDb database, StructuredQuery query, string parentPath)
         {
             Database = database;
-            _query = query;
+            QueryProto = query;
             ParentPath = parentPath;
         }
+
+        /// <summary>
+        /// Specifies the field paths to return in the results.
+        /// </summary>
+        /// <remarks>
+        /// This call replaces any previously-specified projections in the query.
+        /// </remarks>
+        /// <param name="fieldPaths">The dot-separated field paths to select. Must not be null or empty, or contain null or empty
+        /// elements.</param>
+        /// <returns>A new query based on the current one, but with the specified projection applied.</returns>
+        public Query Select(params string[] fieldPaths)
+        {
+            return Select(ConvertFieldPaths(fieldPaths));
+        }
+
+        /// <summary>
+        /// Specifies the field paths to return in the results.
+        /// </summary>
+        /// <remarks>
+        /// This call replaces any previously-specified projections in the query.
+        /// </remarks>
+        /// <param name="fieldPaths">The field paths to select. Must not be null or empty, or contain null elements.</param>
+        /// <returns>A new query based on the current one, but with the specified projection applied.</returns>
+        public Query Select(params FieldPath[] fieldPaths)
+        {
+            GaxPreconditions.CheckNotNull(fieldPaths, nameof(fieldPaths));
+            GaxPreconditions.CheckArgument(fieldPaths.Length != 0, nameof(fieldPaths), "Projection must specify at least one field");
+            GaxPreconditions.CheckArgument(!fieldPaths.Contains(null), nameof(fieldPaths), "Field paths must not contain a null element");
+            var query = QueryProto.Clone();
+            query.Select = new Projection { Fields = { fieldPaths.Select(fp => fp.ToFieldReference()) } };
+            return WithQuery(query);
+        }
+
+        // TODO: Choices...
+        // - Use an enum instead of strings? Rename the middle parameter to "operator"?
+        // - Rename the "op" parameter?
+        // - Rename "Where" to "AddFilter"?
+        // - Reimplement as individual methods, two per operator (accepting string or FieldPath)?
```

Comment:
nit: this appears to be done? That is, there are two Where() methods, one takes a string, t'other takes a FieldPath?

Reply:
Sorry, I'll clarify the TODO - I meant "a method for each query operator", e.g. WhereEqual, WhereLessThan etc.